src/stream/stream/flat_map.rs

```diff
@@ -51,15 +51,22 @@ where
         let mut this = self.project();
         loop {
             if let Some(inner) = this.inner_stream.as_mut().as_pin_mut() {
-                if let item @ Some(_) = futures_core::ready!(inner.poll_next(cx)) {
-                    return Poll::Ready(item);
+                let next_item = futures_core::ready!(inner.poll_next(cx));
+
+                if next_item.is_some() {
+                    return Poll::Ready(next_item);
+                } else {
+                    this.inner_stream.set(None);
```

Comment:
I think if use match, It code more readable. What do you think?
```suggestion
                match futures_core::ready!(inner.poll_next(cx)) {
                    item @ Some(_) => return Poll::Ready(item),
                    None => this.inner_stream.set(None),
```